modules/browser_object_navigation.py

```diff
@@ -0,0 +1,109 @@
+from pypom import Page
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class Navigation(Page):
+    """Page Object Model for nav buttons and AwesomeBar"""
+
+    URL_TEMPLATE = "about:blank"
+    BROWSER_MODES = {
+        "Bookmarks": "*",
+        "Tabs": "%",
+        "History": "^",
+        "Actions": ">",
+    }
+
+    _awesome_bar = (By.ID, "urlbar-input")
+    _xul_source_snippet = (
+        'xmlns:xul="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"'
+    )
+
+    # Tab-to-search
+    _tab_to_search_text_span = (
+        By.CLASS_NAME,
+        "urlbarView-dynamic-onboardTabToSearch-text-container",
+    )
+    _search_mode_span = (By.ID, "urlbar-search-mode-indicator-title")
+
+    # Overflow menu
+    _overflow_item = (
+        By.CSS_SELECTOR,
+        '[class="urlbarView-title urlbarView-overflowable"]',
+    )
+
+    # Search Mode One-Offs
+    _search_one_off_settings_button = (By.ID, "urlbar-anon-search-settings")
+
+    def ensure_chrome_context(self):
+        if self._xul_source_snippet not in self.driver.page_source:
+            self.driver.set_context(self.driver.CONTEXT_CHROME)
+
+    def resume_content_context(self):
+        if self._xul_source_snippet in self.driver.page_source:
+            self.driver.set_context(self.driver.CONTEXT_CONTENT)
+
+    def get_awesome_bar(self) -> WebElement:
+        self.ensure_chrome_context()
+        self.awesome_bar = self.driver.find_element(*self._awesome_bar)
+        return self
+
+    def type_in_awesome_bar(self, term: str) -> WebElement:
+        self.get_awesome_bar()
+        self.awesome_bar.click()
+        self.awesome_bar.send_keys(term)
+        return self
+
+    def set_search_mode_via_awesome_bar(self, mode: str) -> WebElement:
+        if mode in self.BROWSER_MODES:
+            abbr = self.BROWSER_MODES[mode]
+        else:
+            abbr = mode.lower()[:2]
+        self.type_in_awesome_bar(abbr)
+        self.wait.until(EC.visibility_of_element_located(self._tab_to_search_text_span))
+        self.awesome_bar.send_keys(Keys.TAB)
```

Comment:
Why are you not using `type_in_awesome_bar`?

Reply:
This is specifically because I don't want to send another `click` to the bar.